src/content_delegate.js

```diff
@@ -545,7 +545,16 @@ ContentDelegate.prototype.onDownloadAllSubmit = async function (event) {
     const page = document.createElement('html');
     page.innerHTML = html;
     const frames = page.querySelectorAll('iframe');
-    return frames[0].src;
+    if (frames.length) {
+      return frames[0].src;
+    }
+    // Try to extract the PDF URL from the HTML
+    const showTempURL = html.match(/\/cgi-bin\/show_temp.pl?(.*)/);
```

Comment:
(While I'm otherwise here…

I am tempted to write this as

```
    const showTempURL = html.match(new Regexp('/cgi-bin/show_temp.pl?(.*)'));
```
to avoid escaping the forward slashes, but I'm not sure it's actually cleaner.
)

Reply:
🤔 I used a regular expression literal because according to Eslint, it is considered the best practice.

This is the Eslint recommendation: 

> When a regular expression is known in advance, it is considered a best practice to avoid the string literal notation on top of the regular expression notation, and use regular expression literals instead of the constructor function.

You can read more about using this [link](https://eslint.org/docs/latest/rules/prefer-regex-literals#:~:text=When%20a%20regular%20expression%20is,instead%20of%20the%20constructor%20function.).